c/tskit/tables.c

```diff
@@ -7731,26 +7731,42 @@ tsk_table_collection_free(tsk_table_collection_t *self)
  * this does *not* consider the indexes, since these are derived from the
  * tables. We do not consider the file_uuids either, since this is a property of
  * the file that set of tables is stored in. */
+bool
+tsk_table_collection_equivalent(
+    tsk_table_collection_t *self, tsk_table_collection_t *other, tsk_flags_t options)
+{
+    bool cmp_metadata = !!(options & TSK_EQ_CMP_METADATA);
+    bool cmp_provenance = !!(options & TSK_EQ_CMP_PROVENANCE);
+    bool ret
+        = self->sequence_length == other->sequence_length
+          && (!cmp_metadata
+                 || (self->metadata_length == other->metadata_length
```

Comment:
This part isn't compiling on `gcc 9.3.0` as it is asking for brackets around the second half of the `||`

Reply:
yes, solved it in a local commit! thansk!